packages/remark-iframes/src/index.js

```diff
@@ -155,6 +103,59 @@ module.exports = function plugin (opts) {
     if (!visitors) return
     visitors.iframe = (node) => `!(${node.src})`
   }
+
+  return async function transform (tree, vfile, next) {
+    let toVisit = 0
+    visit(tree, 'iframe', async (node) => {
+      toVisit++
+    })
+
+    function nextVisitOrBail () {
+      if (toVisit === 0) next()
+    }
+    nextVisitOrBail()
```

Comment:
The transform only returns when we call its `next` callback. We want to make sure we processed all iframes before calling `next`. For this reason we first count how many iframes we have to process, and only when all iframes have been processed (ie. `visit === 0`) we call `next`.